pkg/controllers/user/setting/register.go

```diff
@@ -0,0 +1,173 @@
+package setting
+
+import (
+	"context"
+	"fmt"
+
+	v1 "github.com/rancher/types/apis/core/v1"
+	v3 "github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// SettingProviderController listens for setting CUD in management API
+// and propagates the changes to the secret of cattle-system namespace in cluster API
+type Controller struct {
+	secrets                 v1.SecretInterface
+	secretsLister           v1.SecretLister
+	managementSettingLister v3.SettingLister
+	clusterName             string
+}
+
+const (
+	CattleNamespace          = "cattle-system"
```

Comment:
most likely we have cattle-system namespace defined in the constants somewhere, think we can reuse it? same with the local cluster name